pkg/controller/installation/products/mobiledeveloperconsole/reconciler.go

```diff
@@ -0,0 +1,246 @@
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/marketplace"
+	"github.com/integr8ly/integreatly-operator/pkg/controller/installation/products/config"
+	"github.com/integr8ly/integreatly-operator/pkg/resources"
+	appsv1 "github.com/openshift/api/apps/v1"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultInstallationNamespace = "mdc"
+	defaultSubscriptionName      = "integreatly-mobile-developer-console"
+	resourceName                 = "mobiledeveloperconsole"
+	routeResourceName            = "mobiledeveloperconsole-mdc-proxy"
+)
+
+type Reconciler struct {
+	Config        *config.MobileDeveloperConsole
+	ConfigManager config.ConfigReadWriter
+	mpm           marketplace.MarketplaceInterface
+	logger        *logrus.Entry
+	*resources.Reconciler
+}
+
+func NewReconciler(configManager config.ConfigReadWriter, instance *v1alpha1.Installation, mpm marketplace.MarketplaceInterface) (*Reconciler, error) {
+	config, err := configManager.ReadMobileDeveloperConsole()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read mobile developer console")
+	}
+
+	if config.GetNamespace() == "" {
+		config.SetNamespace(instance.Spec.NamespacePrefix + defaultInstallationNamespace)
+	}
+
+	err = config.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "mdc config is not valid")
+	}
+
+	logger := logrus.WithFields(logrus.Fields{"product": config.GetProductName()})
+
+	return &Reconciler{
+		ConfigManager: configManager,
+		Config:        config,
+		mpm:           mpm,
+		logger:        logger,
+		Reconciler:    resources.NewReconciler(mpm),
+	}, nil
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation, product *v1alpha1.InstallationProductStatus, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+
+	phase, err := r.ReconcileNamespace(ctx, r.Config.GetNamespace(), inst, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	version, err := resources.NewVersion(v1alpha1.OperatorVersionMDC)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "invalid version number for mdc")
+	}
+	phase, err = r.ReconcileSubscription(
+		ctx,
+		inst,
+		marketplace.Target{Pkg: defaultSubscriptionName, Channel: marketplace.IntegreatlyChannel, Namespace: r.Config.GetNamespace()},
+		client,
+		version,
+	)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileCustomResource(ctx, client, inst)
```

Comment:
```suggestion
	phase, err = r.reconcileCustomResource(ctx, client, inst)
	phase, err = r.reconcileComponents(ctx, client, inst)
```

to standardise on method names in the product reconcilers mainly